packages/data/src/macros/macros_oot.yml

```
@@ -151,6 +151,7 @@
 "has_sunlight_arrows": "setting(sunlightArrows) && can_use_light_arrows"
 "has_fire_arrows": "can_use_bow && (has(ARROW_FIRE) || has(SHARED_ARROW_FIRE)) && has_magic"
 "has_lens_strict": "has_magic && (has(LENS) || has(SHARED_LENS))"
+"has_spin_upgrade": "cond(setting(sharedSpinUpgrade), has(SHARED_SPIN_UPGRADE), has(SPIN_UPGRADE))"
```

Oversight on my part. OOT can have the spin upgrade by just having magic without the spin upgrade item enabled, so would need slightly altered.

```suggestion
"has_spin_upgrade": "cond(setting(spinUpgradeOot), cond(setting(sharedSpinUpgrade), has(SHARED_SPIN_UPGRADE), has(SPIN_UPGRADE)), true)"
```

---

Changed.